src/suites/cts/validation/render_pass_descriptor.spec.ts

```diff
@@ -0,0 +1,554 @@
+    });
+  });
+
+  const descriptor: GPURenderPassDescriptor = {
+    colorAttachments: [colorAttachment],
+  };
+
+  await t.tryRenderPass(false, descriptor);
+});
+
+g.test('use of multisampled attachment and non multisampled resolve target is allowed', async t => {
+  const multisampledColorTexture = t.createTexture({ sampleCount: 4 });
+  const resolveTargetTexture = t.createTexture({ sampleCount: 1 });
+
+  const colorAttachment = t.getColorAttachment(multisampledColorTexture);
+  colorAttachment.resolveTarget = resolveTargetTexture.createView();
+
+  const descriptor: GPURenderPassDescriptor = {
+    colorAttachments: [colorAttachment],
+  };
+
+  t.tryRenderPass(true, descriptor);
+});
+
+g.test('use a resolve target in a format different than the attachment is not allowed', async t => {
+  const multisampledColorTexture = t.createTexture({ sampleCount: 4 });
+  const resolveTargetTexture = t.createTexture({ format: 'bgra8unorm' });
+
+  const colorAttachment = t.getColorAttachment(multisampledColorTexture);
+  colorAttachment.resolveTarget = resolveTargetTexture.createView();
+
+  const descriptor: GPURenderPassDescriptor = {
+    colorAttachments: [colorAttachment],
+  };
+
+  await t.tryRenderPass(false, descriptor);
+});
+
+g.test('size of the resolve target must be the same as the color attachment', async t => {
+  const size = 16;
+  const multisampledColorTexture = t.createTexture({ width: size, height: size, sampleCount: 4 });
+  const resolveTargetTexture = t.createTexture({
+    width: size * 2,
+    height: size * 2,
+    mipLevelCount: 2,
+  });
+
+  {
+    const resolveTargetTextureView = resolveTargetTexture.createView({ baseMipLevel: 0 });
+
+    const colorAttachment = t.getColorAttachment(multisampledColorTexture);
+    colorAttachment.resolveTarget = resolveTargetTextureView;
+
+    const descriptor: GPURenderPassDescriptor = {
+      colorAttachments: [colorAttachment],
+    };
+
+    await t.tryRenderPass(false, descriptor);
+  }
+  {
+    const resolveTargetTextureView = resolveTargetTexture.createView({ baseMipLevel: 1 });
+
+    const colorAttachment = t.getColorAttachment(multisampledColorTexture);
+    colorAttachment.resolveTarget = resolveTargetTextureView;
+
+    const descriptor: GPURenderPassDescriptor = {
+      colorAttachments: [colorAttachment],
+    };
+
+    t.tryRenderPass(true, descriptor);
+  }
+});
+
+g.test('check depth stencil attachment sample counts mismatch', async t => {
+  const multisampledDepthStencilTexture = t.createTexture({
+    sampleCount: 4,
+    format: 'depth24plus-stencil8',
+  });
+
+  {
+    // It is not allowed to use a depth stencil attachment whose sample count is different from the
+    // one of the color attachment
+    const depthStencilTexture = t.createTexture({
+      sampleCount: 1,
+      format: 'depth24plus-stencil8',
+    });
+    const descriptor: GPURenderPassDescriptor = {
+      colorAttachments: [t.getColorAttachment(depthStencilTexture)],
```

Comment:
this is setting a depth stencil texture as a color attachment, which I don't think is the intent of this test

Reply:
Thank you!